util/src/main/scala/com/scalableminds/util/io/ZipIO.scala

```diff
@@ -145,6 +145,27 @@ object ZipIO extends LazyLogging {
     }
   }
 
+  def tryGunzip(possiblyCompressed: Array[Byte]): Array[Byte] =
+    tryo(gunzip(possiblyCompressed)).toOption.getOrElse(possiblyCompressed)
+
+  def gunzip(compressed: Array[Byte]): Array[Byte] = {
+    val is = new GZIPInputStream(new ByteArrayInputStream(compressed))
+    val os = new ByteArrayOutputStream()
+    try {
+      val buffer = new Array[Byte](1024)
```

Comment:
Any reason for this specific buffer length?

Reply:
I just stole it from the gzip method. Since all bytes should already be in memory in this case, the buffer size matters little. I believe powers of two are recommended if the data source or destination is on disk, to match whole multiples of the disk block size, but here it does not really matter that much. If you have any more insight about the ideal value here, please let me know :)